modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -359,13 +366,18 @@ sda.enkf.multisite <- function(settings,
         restart.list <-
           furrr::future_pmap(list(out.configs, conf.settings %>% `class<-`(c("list")), params.list, inputs.split),
                              function(configs, settings, new.params, inputs) {
+                               #if the new state for each site only has one row/col.
+                               #then we need to convert it to matrix to solve the indexing issue.
+                               new_state <- new.state[, which(attr(X, "Site") %in% settings$run$site$id)]
```

Comment:
This might work fine because of different scopes, but to me it feels a bit dangerous and simpler to just use a different variable name in the assignment.

Reply:
I don't understand what you mean here.

Reply:
new_state isn't an argument to the function, so you're accessing a global variable, which you're then modifying (subsetting) within the function. In such situations it can be ambiguous whether those changes are local or global, and whether you WANT the changes to be local or global. I think you want the changes to be only local. doing something like `new_state_site = new_site[, which(attr(X, "Site") %in% settings$run$site$id)]` would be more clear and less likely to cause errors related to scope.

Reply:
Fixed.